blocks/src/custom-status/block.js

```diff
@@ -14,47 +14,56 @@ let { SelectControl } = wp.components;
 let statuses = window.EditFlowCustomStatuses.map( s => ({ label: s.name, value: s.slug }) );
 
 /**
- * Hack :(
- *
- * @see https://github.com/WordPress/gutenberg/issues/3144
- *
- * Gutenberg overrides the label of the Save button after save (i.e. "Save Draft"). But there's no way to subscribe to a "post save" message.
- *
- * So instead, we're keeping the button label generic ("Save"). There's a brief period where it still flips to "Save Draft" but that's something we need to work upstream to find a good fix for.
+ * Subscribe to changes so we can set a default status and update a button's text.
  */
-let sideEffectL10nManipulation = () => {
-  let node = document.querySelector('.editor-post-save-draft');
-  if ( node ) {
-    document.querySelector( '.editor-post-save-draft' ).innerText = `${ __( 'Save' ) }`
-  }
-}
-
-// Set the status to the default custom status.
+let buttonTextObserver = null;
 subscribe( function () {
-  const postId = select( 'core/editor' ).getCurrentPostId();
-  // Post isn't ready yet so don't do anything.
-  if ( ! postId ) {
-    return;
-  }
-
-  // For new posts, we need to force the our default custom status.
-  // Otherwise WordPress will force it to "Draft".
-  const isCleanNewPost = select( 'core/editor' ).isCleanNewPost();
-  if ( isCleanNewPost ) {
-    dispatch( 'core/editor' ).editPost( {
-      status: ef_default_custom_status
-    } );
-
-    return;
```

Comment:
Not sure why this `return` was here in the first place, but why drop it?

Reply:
I'll do some testing around this, but logically felt like it should be removed in case the button needs updating / an observer adding during that subscribe().

Reply:
Doesn't seem to matter really - this dispatch is sent immediately when a new post is loaded, before the save button is visible.

If core were to show the save button, but perhaps disabled, then we would want the logic to "fallthrough" like it currently does. So the current implementation I guess is just a little more prepared if anything, but ultimately doesn't matter much since the subsequent checks are very fast, and they are 10 bajillion subscribe() calls anyways 😆